src/main/scala/org/broadinstitute/dsde/vault/services/analysis/UpdateServiceHandler.scala

```diff
@@ -56,7 +65,12 @@ case class UpdateServiceHandler(requestContext: RequestContext, dmService: Actor
 
     case BossObjectCreated(bossObject: BossCreationObject, creationKey: String) =>
       bossObjects += creationKey -> bossObject.objectId.get
-      bossService ! BossClientService.BossResolveObject(BossDefaults.getResolutionRequest("PUT"), bossObject.objectId.get, creationKey)
+      // If the client is using the "X-Force-Location" header, no need to populate PUT urls.
+      // Instead, forward to the DM service if all BOSS objects have been created.
+      if (!forceLocationHeader)
+        bossService ! BossClientService.BossResolveObject(BossDefaults.getResolutionRequest("PUT"), bossObject.objectId.get, creationKey)
+      if (fileCount == bossObjects.size)
+        dmService ! DmClientService.DMUpdateAnalysis(dmId.get, new AnalysisUpdate(providedFiles))
 
     case BossObjectResolved(bossObject: BossResolutionResponse, creationKey: String) =>
       bossURLs += creationKey -> bossObject.objectUrl
```

Comment:
Agree with David.  I didn't see this before I made my comment below.  It's equivalent to what you're saying.